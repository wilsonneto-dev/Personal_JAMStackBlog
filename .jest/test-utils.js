import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import defaultTheme from 'styles/themes/default';

const WithTheme = ({ children }) => {
  return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: WithTheme, ...options });
