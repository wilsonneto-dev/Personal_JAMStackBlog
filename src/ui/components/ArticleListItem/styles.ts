import styled from 'styled-components';

export const Wrapper = styled.div``;

export const Article = styled.article`
  font-family: 'Roboto', sans-serif;
  width: 100%;
  min-width: 25rem;
  max-width: 35rem;
  display: inline-block;
  padding: 0 0.5rem;
  margin: 0 auto;
  margin-bottom: 4rem;

  @media (max-width: 600px) {
    width: 100%;
    margin: 0 auto 4rem auto;
    max-width: 35rem;
  }

  header {
    img {
      width: 100%;
      padding: 0;
      margin: 0 0 0.5rem;
      border-radius: ${({ theme }) => theme.sizes.borderRadius};
    }

    h1 {
      font-size: 2rem;
      font-weight: normal;
      transition: all 0.25s;
      color: ${({ theme }) => theme.colors.textDarken};
    }
  }

  &:hover {
    header h1 {
      color: ${({ theme }) => theme.colors.featuredBackground};
    }
  }

  p {
    color: ${({ theme }) => theme.colors.text};
    font-size: 1.5rem;
    padding: 1.5rem 0;
  }

  footer {
    color: ${({ theme }) => theme.colors.textLight};
    font-size: 1.5rem;

    strong {
      color: ${({ theme }) => theme.colors.textDarken};
    }
  }
`;
