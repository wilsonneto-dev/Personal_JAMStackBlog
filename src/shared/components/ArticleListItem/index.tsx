import Link from 'next/link';
import IArticle from 'models/interfaces/IArticle';

import * as S from './styles';

export type IArticleListItem = Pick<
  IArticle,
  'thumbURL' | 'title' | 'description' | 'author' | 'datetime' | 'datetext'
>;

const ArticleListItem = ({
  thumbURL,
  title,
  description,
  author,
  datetext,
  datetime
}: IArticleListItem) => (
  <S.Article>
    <header>
      <Link href="/articles/[category]/[slug]" as="/articles/react/test01">
        <a>
          <img src={thumbURL} />
        </a>
      </Link>
      <Link href="/articles/[category]/[slug]" as="/articles/react/test01">
        <a>
          <h1>{title}</h1>
        </a>
      </Link>
    </header>
    <p>
      <Link href="/articles/[category]/[slug]" as="/articles/react/test01">
        <a>{description}</a>
      </Link>
    </p>
    <footer>
      Por <strong>{author}</strong> em{' '}
      <time dateTime={datetime}>{datetext}</time>
    </footer>
  </S.Article>
);

export default ArticleListItem;
