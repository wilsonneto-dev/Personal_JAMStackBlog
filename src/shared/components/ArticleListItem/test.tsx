import { render, screen } from '@testing-library/react';
import { renderWithTheme } from '../../../../.jest/test-utils';

import ArticleListItem, { IArticleListItem } from '.';

const exampleData: IArticleListItem = {
  thumbURL: '/img/350x160.png',
  title: 'Atualização aplicações React Native para a última versao',
  description:
    'Atualização aplicações React Native para a última versao Atualização' +
    ' aplicações React Atualização aplicações React Native para a última' +
    ' versao Atualização aplicações React',
  author: 'Wilson Neto',
  datetime: '2020-11-01 20:00',
  datetext: '01/11/2020'
};

describe('<ArticleListItem />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<ArticleListItem {...exampleData} />);

    expect(screen.getByText('teste de artigo 01')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
