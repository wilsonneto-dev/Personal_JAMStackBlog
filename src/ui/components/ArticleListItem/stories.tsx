import { Story, Meta } from '@storybook/react/types-6-0';
import ArticleListItem, { IArticleListItem } from '.';

const Template: Story<IArticleListItem> = (args) => (
  <ArticleListItem {...args} />
);

export default {
  title: 'Articles/ArticleListItem',
  component: ArticleListItem
} as Meta;

export const Default = Template.bind({});

Default.args = {
  thumbURL: 'https://source.unsplash.com/random/350x160',
  title: 'Atualização aplicações React Native para a última versao',
  description:
    'Atualização aplicações React Native para a última versao Atualização aplicações React',
  author: 'Wilson Neto',
  datetime: '2020-11-01 20:00',
  datetext: '01/11/2020'
};
