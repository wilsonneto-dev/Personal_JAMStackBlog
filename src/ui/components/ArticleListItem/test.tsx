import { screen } from '@testing-library/react';
import { renderWithTheme } from '../../../../.jest/test-utils';

import ArticleListItem, { IArticleListItem } from '.';

const exampleData: IArticleListItem = {
  thumbURL: '/img/350x160.png',
  title: 'Atualização aplicações React Native para a última versao',
  description:
    'Atualização aplicações React Native para a última versao Atualização' +
    ' aplicações React Atualização aplicações React Native para a última' +
    ' versao Atualização aplicações React',
  author: 'Wilson Neto',
  datetime: '2020-11-01 20:00',
  datetext: '01/11/2020'
};

describe('<ArticleListItem />', () => {
  it('should render the heading, text and footer', () => {
    renderWithTheme(<ArticleListItem {...exampleData} />);

    expect(screen.getByText(exampleData.title)).toBeInTheDocument();

    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getAllByRole('link').length).toBeGreaterThan(0);
    expect(screen.getByRole('contentinfo')).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = renderWithTheme(<ArticleListItem {...exampleData} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
