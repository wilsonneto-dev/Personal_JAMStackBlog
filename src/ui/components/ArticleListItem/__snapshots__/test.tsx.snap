// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleListItem /> should match the snapshot 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  width: 100%;
  min-width: 25rem;
  max-width: 35rem;
  display: inline-block;
  padding: 0 0.5rem;
  margin: 0 auto;
  margin-bottom: 4rem;
}

.c0 header img {
  width: 100%;
  padding: 0;
  margin: 0 0 0.5rem;
  border-radius: 5px;
}

.c0 header h1 {
  font-size: 2rem;
  font-weight: normal;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  color: #4e5256;
}

.c0:hover header h1 {
  color: #612F74;
}

.c0 p {
  color: #909498;
  font-size: 1.5rem;
  padding: 1.5rem 0;
}

.c0 footer {
  color: #909498;
  font-size: 1.5rem;
}

.c0 footer strong {
  color: #4e5256;
}

@media (max-width:600px) {
  .c0 {
    width: 100%;
    margin: 0 auto 4rem auto;
    max-width: 35rem;
  }
}

<article
  class="c0"
>
  <header>
    <a
      href="/articles/react/test01"
    >
      <img
        src="/img/350x160.png"
      />
    </a>
    <a
      href="/articles/react/test01"
    >
      <h1>
        Atualização aplicações React Native para a última versao
      </h1>
    </a>
  </header>
  <p>
    <a
      href="/articles/react/test01"
    >
      Atualização aplicações React Native para a última versao Atualização aplicações React Atualização aplicações React Native para a última versao Atualização aplicações React
    </a>
  </p>
  <footer>
    Por 
    <strong>
      Wilson Neto
    </strong>
     em
     
    <time
      datetime="2020-11-01 20:00"
    >
      01/11/2020
    </time>
  </footer>
</article>
`;
